app/main.py

```diff
@@ -1,25 +1,91 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: SlideLimitationValidator,
+            name: str
+    ) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(
+            self,
+            instance: SlideLimitationValidator,
+            owner: SlideLimitationValidator
```

Comment:
Change the type annotation for `instance` and `owner` to `Any` here and in the other methods of the `IntegerRange` class not to limit the `IntegerRange` class use